arch/posix/CMakeLists.txt

```diff
@@ -1,7 +1,15 @@
 # SPDX-License-Identifier: Apache-2.0
 
+# This is pretty x86 centric. Some more generic method would be needed.
```

Comment:
Yep. I think it would be nicer to add a new kconfig choice for the POSIX arch, something like `POSIX_ARCH_TARGET`
which could start with `POSIX_ARCH_TARGET{_X86,_X86_64}` and default to `POSIX_ARCH_TARGET_X86` if not `64BIT` (or have 64 bit one be selected by `NATIVE_POSIX_64BIT`).

Then this if below could be based on it instead of `64BIT`.

Whenever we start supporting other host targets we'd add new `POSIX_ARCH_TARGET`s and add new compile options here for them.

Reply:
Postponed